### ISSUE TITLE
Add Ability to Follow Mastodon Users

### ISSUE THREAD

**Issue by stuartbreckenridge** [Ranchero-Software/NetNewsWire#3748]
How:

- Add a "Follow Mastodon User" option to the + button on the Master Feeds screen
- This will present UI that will ask the user for the username to follow in the format of <username>@<mastodon.domain>
- NNW will convert this to the RSS url: `https://<mastodon.domain>/users/ <username>.rss` and check to make sure it's valid
- if it's valid, it will add to the account / folder selected

**Comment by stuartbreckenridge** [comment]
@brentsimmons Is this what you had in mind?

https://user-images.githubusercontent.com/7046652/203200831-da2743e7-94fe-41af-b31b-f2842ddbaca9.mov

**Comment by alltom** [comment]
fyi, RSS feeds don't include boosts: https://github.com/mastodon/mastodon/issues/11290

Is there another protocol endpoint that would make NNW usable as a Mastodon client?

**Comment by stuartbreckenridge** [comment]
The aim isn’t to make NetNewsWire a Mastodon client, but instead to make it easier to follow Mastodon RSS feeds.

**Comment by alltom** [comment]
Yeah, but boosts are part of feeds. I got here because I noticed the RSS feeds of some of the people I followed were weirdly empty, and it turned out because most of their feeds were boosts.

NNW's Reddit support doesn't even use RSS, so I didn't think that would be a blocker here…

**Comment by vincode-io** [comment]
@alltom you are correct. NNW uses the Reddit API, not RSS for its support. I don’t think we can rule out using ActivityPub for a Mastodon Extension in the future.
